shared/utility/math/coordinates.h

```diff
@@ -50,6 +50,22 @@ namespace math {
             return result;
         }
 
+        template <typename T, typename U = typename T::Scalar>
+        inline Eigen::Matrix<U, 3, 1> sphericalToCartesian(const Eigen::MatrixBase<T>& sphericalCoordinates) {
+            U distance  = sphericalCoordinates[0];
+            U cos_theta = cos(sphericalCoordinates[1]);
+            U sin_theta = sin(sphericalCoordinates[1]);
+            U cos_phi   = cos(sphericalCoordinates[2]);
+            U sin_phi   = sin(sphericalCoordinates[2]);
+            Eigen::Matrix<U, 3, 1> result;
+
+            result[0] = distance * cos_theta * cos_phi;
+            result[1] = distance * sin_theta * cos_phi;
+            result[2] = distance * sin_phi;
```

Comment:
Should this be `y = distance * sin_theta * sin_phi`, and `z = distance * cos_phi`?

Reply:
A little googling and I think you're right. I just copied the arma function with the same logic, just above that one. I will fix them both now.

Reply:
actually, isn't `x = distance * sin_phi * cos_theta` too?

Reply:
Yeah that too. @Bidski we're not missing something are we
is this function used anywhere?

Reply:
You could be, depending on your definition of `phi` and `theta` and `x` `y` and `z` the answer from Wikipedia might be wrong